pkg/virt-controller/watch/migration.go

```diff
@@ -293,15 +305,31 @@ func copy(migration *kubev1.Migration) (*kubev1.Migration, error) {
 }
 
 // Returns the number of  running pods and if a pod for exactly that migration is currently running
-func investigateTargetPodSituation(migration *kubev1.Migration, podList *k8sv1.PodList) (int, *k8sv1.Pod) {
+func investigateTargetPodSituation(migration *kubev1.Migration, podList *k8sv1.PodList, migrationStore cache.Store) (int, *k8sv1.Pod, error) {
 	var targetPod *k8sv1.Pod = nil
-	for _, pod := range podList.Items {
+	podCount := 0
+	for idx, pod := range podList.Items {
 		if pod.Labels[kubev1.MigrationUIDLabel] == string(migration.GetObjectMeta().GetUID()) {
-			targetPod = &pod
-			break
+			targetPod = &podList.Items[idx]
+			podCount += 1
+			continue
+		}
+		key := fmt.Sprintf("%s/%s", pod.ObjectMeta.Namespace, pod.Labels[kubev1.MigrationLabel])
```

Comment:
Could you also check for the existence of the label? If it does not have this label, it is the original pod, which can't be a target pod by definition. Then we have everything consistent, and will ever only report real target pods from active migrations. This means, we can consistently handle the result of this function with:
if no target pod:
 * count >=1: another migration is going on, fail
 * count ==0: we have to create a target pod, do so and continue